src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_pending_resumption_handler.cc

```diff
@@ -31,252 +31,197 @@
+uint32_t get_corr_id_from_message(const smart_objects::SmartObject& message) {
+  using namespace application_manager;
+  return message[strings::params][strings::correlation_id].asInt();
 }
+CREATE_LOGGERPTR_GLOBAL(logger_, "VehicleInfoPendingResumptionHandler")
 
-void VehicleInfoPendingResumptionHandler::RemoveSucessfulSubscriptions(
-    std::set<std::string>& subscriptions,
-    std::set<std::string>& successful_subscriptions) {
-  for (auto subscription : subscriptions) {
-    if (helpers::in_range(successful_subscriptions, subscription)) {
-      subscriptions.erase(subscription);
+std::set<std::string> SubscriptionsFromResponse(
+    const smart_objects::SmartObject& response,
+    std::function<bool(const smart_objects::SmartObject& vehicle_data)>
+        vehicle_data_check) {
+  using namespace application_manager;
+  std::set<std::string> result;
+  const auto response_params = response[strings::msg_params];
+  const auto response_keys = response_params.enumerate();
+  for (auto key : response_keys) {
+    if (vehicle_data_check(response_params[key])) {
+      result.insert(key);
     }
   }
+  return result;
 }
 
-void VehicleInfoPendingResumptionHandler::ClearPendingRequestsMap() {
+std::set<std::string> SuccessfulSubscriptionsFromResponse(
+    const smart_objects::SmartObject& response) {
+  LOG4CXX_AUTO_TRACE(logger_);
   using namespace application_manager;
 
-  for (auto const& it : pending_requests_) {
-    const hmi_apis::FunctionID::eType timed_out_pending_request_fid =
-        static_cast<hmi_apis::FunctionID::eType>(
-            it.second[strings::params][strings::function_id].asInt());
-    unsubscribe_from_event(timed_out_pending_request_fid);
+  auto succesfull_response = [](const smart_objects::SmartObject& response) {
+    const auto result_code = static_cast<hmi_apis::Common_Result::eType>(
+        response[strings::params][application_manager::hmi_response::code]
+            .asInt());
+    return (result_code == hmi_apis::Common_Result::SUCCESS ||
+            result_code == hmi_apis::Common_Result::WARNINGS);
+  };
+
+  std::set<std::string> result;
+  if (!succesfull_response(response)) {
+    return result;
   }
 
-  pending_requests_.clear();
+  auto sucessful_vehicle_data =
+      [](const smart_objects::SmartObject& vehicle_data) {
+        constexpr auto kSuccess =
+            hmi_apis::Common_VehicleDataResultCode::VDRC_SUCCESS;
+        const auto vd_result_code = vehicle_data[strings::result_code].asInt();
+        return kSuccess == vd_result_code;
+      };
+  return SubscriptionsFromResponse(response, sucessful_vehicle_data);
 }
 
+VehicleInfoPendingResumptionHandler::VehicleInfoPendingResumptionHandler(
+    application_manager::ApplicationManager& application_manager)
+    : ExtensionPendingResumptionHandler(application_manager) {}
+
 void VehicleInfoPendingResumptionHandler::ClearPendingResumptionRequests() {
   LOG4CXX_AUTO_TRACE(logger_);
+  TriggerPendingResumption();
```

Comment:
I like this approach, now this operation became more clearer, than untransparent and complicated deleting something, as before